src/ads/MobileAd.ts

```diff
@@ -76,19 +88,44 @@ export class MobileAd {
       this._isLoadCalled = false;
     }
 
-    if (this._onAdEventHandler) {
-      let nativeError;
-      if (error) {
-        nativeError = NativeError.fromEvent(error, 'googleMobileAds');
-      }
+    let payload: AdEventPayload<EventType> = data;
+    if (error) {
+      payload = NativeError.fromEvent(error, 'googleMobileAds');
+    }
+    this._adEventsListeners.forEach(listener => {
+      listener({
+        type,
+        payload,
+      });
+    });
+    this._getAdEventListeners(type).forEach(listener => {
+      listener(payload);
+    });
+  }
+
+  protected _addAdEventsListener(listener: AdEventsListener) {
+    const index = this._adEventsListeners.push(listener as AdEventsListener<EventType>) - 1;
+    return () => {
+      this._adEventsListeners.splice(index, 1);
+    };
+  }
+
+  protected _addAdEventListener(type: EventType, listener: AdEventListener) {
+    const index = this._getAdEventListeners(type).push(listener) - 1;
+    return () => {
+      this._getAdEventListeners(type).splice(index, 1);
```

Comment:
same question here about splicing

I guess it could do with a test that probes the two cases:

"array overrun / unsub early listener then unsub last listner" case:
- you add more than one listener,
- remove one of the earlier listeners
- then trigger an event and make sure all correct listeners are called
- remove the listener added last to make sure it doesn't error
- trigger an event and make sure only remaining listeners are called

"array ordinal shift tracking case"
- add 3 listeners make sure all trigger
- unsub 1st and 2nd listener, make sure 3rd triggers and 1st / 2ndd do not